zebra/zebra_rnh.c

```diff
@@ -515,8 +515,9 @@ static bool rnh_check_re_nexthops(const struct route_entry *re,
 
 	/* Some special checks if registration asked for them. */
 	if (CHECK_FLAG(rnh->flags, ZEBRA_NHT_CONNECTED)) {
-		if ((re->type == ZEBRA_ROUTE_CONNECT)
-		    || (re->type == ZEBRA_ROUTE_STATIC))
+		if ((re->type == ZEBRA_ROUTE_CONNECT) ||
+		    (re->type == ZEBRA_ROUTE_STATIC) ||
+		    (re->type == ZEBRA_ROUTE_LOCAL))
```

Comment:
This is another place with unclear semantics. The local address is not really usable if you need a nexthop for something. Is this a new way to configure directly connected networks, by using a local address as nexthop?

⇒ intended behavior needs to be written down

Reply:
removed local and added comments about intended behavior to the code